ENH: `read_mmcr` method to read ARM MMCR b1-level data files

isilber opened the pull request:
A method to robustly read ARM MMCR b1-level data files (the datastream available through the ARM Archive).
The method returns a `dict` with keys representing each of the operated MMCR modes (available in the data files) and values representing the corresponding `Radar` objects. An optional input parameter (`mode_to_extract`) enables only extracting a single mode, in which case a single `Radar` object is returned.

The method is written in a similar format to the `read_kazr` method (which has some clutter in it) under the same `aux_io` module.
That module lacks unit tests (as also noted in its `__init__.py` script) and therefore I didn't include unit tests.

- [ x] Documentation reflects changes

zssherman commented on pyart/aux_io/arm_vpt.py:
```diff
@@ -218,3 +218,269 @@ def read_kazr(
+        value of None are still included in the radar.fields dictionary, to
+        exclude them use the `exclude_fields` parameter. Fields which are
+        mapped by this dictionary will be renamed from key to value.
+    additional_metadata : dict of dicts, optional
```
I would be fine with removing this here and in the function above and just setting the call below to None if not used.

zssherman replied:
Yeah we can remove both in a future PR.

zssherman reviewed (commented):
I left one comment, besides that looks good. @mgrover1 can give this a review as well, when you have time?

isilber commented:
I just added options to read dual-pol channels' MMCR data (limited number of data files).
I also changed the PRT specifications to be read from the loaded file instead of hardwired handbook values

mgrover1 reviewed (approved):
This looks great! Thanks @isilber for putting this together

zssherman commented:
Nope, merge away! Thanks again @isilber !
